src/NServiceBus.Core.Analyzer.Tests/AwaitOrCaptureTasksAnalyzerTests.cs

```diff
@@ -0,0 +1,181 @@
+                Locations = new[] { new DiagnosticResultLocation("Test0.cs", 6, 9) },
+            };
+
+            await Verify(source, expected);
+        }
+
+        [TestCase("RequestTimeout<TimeoutMessage>(context, DateTime.Now);")]
+        [TestCase("RequestTimeout<TimeoutMessage>(context, DateTime.Now, new TimeoutMessage());")]
+        [TestCase("RequestTimeout<TimeoutMessage>(context, TimeSpan.Zero);")]
+        [TestCase("RequestTimeout<TimeoutMessage>(context, TimeSpan.Zero, new TimeoutMessage());")]
+        [TestCase("ReplyToOriginator(context, new object());")]
+        public async Task DiagnosticsIsReportedForSagaAPIs(string api)
+        {
+            var source =
+                $@"using System;
+using System.Threading.Tasks;
+using NServiceBus;
+class TestSaga : Saga<TestSagaData>, IHandleMessages<TestMessage>
+{{
+    public Task Handle(TestMessage message, IMessageHandlerContext context)
+    {{
+        {api}
+        return Task.FromResult(0);
+    }}
+
+    protected override void ConfigureHowToFindSaga(SagaPropertyMapper<TestSagaData> mapper)
+    {{
+    }}
+}}";
+            var expected = new DiagnosticResult
+            {
+                Id = "NSB0001",
+                Message = "Expression calling an NServiceBus method creates a Task that is not awaited or assigned to a variable.",
+                Severity = DiagnosticSeverity.Error,
+                Locations = new[] { new DiagnosticResultLocation("Test0.cs", 8, 9) },
+            };
+            await Verify(source, expected);
+
+        }
+
+        [TestCase("")]
+        [TestCase(
+@"using NServiceBus;
+using System.Threading.Tasks;
+public class TestHandler : IHandleMessages<TestMessage>
+{
+    public async Task Handle(object message, IMessageHandlerContext context)
+    {
+        context.Send(new object(), new SendOptions());
+        return Task.FromResult(0);
+    }
+}")]
+        [TestCase(
+@"using NServiceBus;
+using System.Threading.Tasks;
+public class TestHandler : IHandleMessages<TestMessage>
+{
+    public async Task Handle(object message, IMessageHandlerContext context)
+    {
+        await context.Send(new object(), new SendOptions());
+    }
+}", Description = "Covered by CS4014")]
+        [TestCase(
+@"using NServiceBus;
+using System.Threading.Tasks;
+public class TestHandler : IHandleMessages<TestMessage>
+{
+    public Task Handle(object message, IMessageHandlerContext context) =>
+        context.Send(new object(), new SendOptions());
+}")]
+        [TestCase(
+            @"using NServiceBus;
+using System.Threading.Tasks;
+public class Program
+{
+    public void SendSync(object message, IMessageSession session)
+    {
+        session.Send(message).GetAwaiter().GetResult();
+    }
+}")]
+        [TestCase(
+            @"using NServiceBus;
+using System.Threading.Tasks;
+public class Program
+{
+    public void SendSync(object message, IMessageSession session)
+    {
+        session.Send(message).ConfigureAwait(false);
```

Comment:
in this case I'm not sure whether we shouldn't handle it. Thoughts @adamralph ?

Reply:
`ConfigureAwait(false)` makes only sense when you await the task returned. This sounds like a really rare case. If more cases needs to be covered I think the time would be better spent on https://github.com/Particular/NServiceBus/pull/5155#issuecomment-388790529

Reply:
yeah fair point. I could see some scenario where users add ConfigureAwait because it's shown in all our samples but don't await it, but I expect that to be fairly rare and happy to ignore this.

Reply:
btw @danielmarbach , `Endpoint.Create/Start` is supported by the analyzer now (Endpoint.Start is already handled by the compiler anyway)

Reply:
> `ConfigureAwait(false)` makes only sense when you await the task returned. This sounds like a really rare case.

I do not think that is common knowledge. I think this should trigger a diagnostic. If we were banking on developers knowing things about using tasks and being able to find them easily we wouldn't even need an analyzer.

Reply:
Btw I did not say we should not cover this case. I only said it might make more sense to cover the other scenarios outlined in my comments first. And by looking at how we defined the dropping the task on the floor scanner this is a valid usage scenario which doesn't drop the task. So in that sense we can argue that if the job of the analyzer is to cover dropped tasks this scenario outlined here is not a dropped task and thus should not triggera warn